### PR TITLE
overhaul dev guide indices, titles, tags

### PR DESCRIPTION
**Pull request by napcs**
- Change Feature Guide to developer guide
- Get titles consistent - drop "feature guide" and "temporal" from pages, normalize the rest of the titles.
- Combine Python Cancellation and Interrupting Workflows pages into one page like the other gudies
- Update tags where appropriate

### CODE REVIEW

File: docs/develop/go/index.mdx [hunk 1]
```diff
@@ -45,101 +45,113 @@ The [Testing suite feature guide](/develop/go/testing-suite) shows how to set up
...
-The [Asynchronous Activity Completion feature guide](/develop/go/asynchronous-activity-completion) shows how to complete Activities asynchronously.
+Interrupt a Workflow Execution with a Cancel or Terminate action.
 
-## Versioning
+- [Handle a Workflow Cancellation Request](/develop/go/cancellation#handle-cancellation-in-workflow): Interrupt a Workflow Execution and its Activities through Workflow cancellation.
+- [Request Cancellation](/develop/go/cancellation#request-cancellation)
 
-The [Versioning feature guide](/develop/go/versioning) shows how to change Workflow Definitions without causing non-deterministic behavior in current long-running Workflows.
+
+## [Asynchronous Activity Completion](/develop/go/asynchronous-activity-completion)
```
[the remark this answers is not in the document]
Comment (fairlydurable) [reply docs/develop/go/index.mdx:72]: Also I left some other comments in PHP (didn't know it was PHP when I was looking at it) that apply  across these)

File: docs/develop/go/observability.mdx [hunk 2]
```diff
@@ -43,7 +40,10 @@ tags:
   - workflow-execution
   - workflow-executions
   - workflows
-description: Explore how to monitor your Temporal Application state using Metrics, Tracing, Logging, and Visibility features. Learn about emitting metrics, configuring tracing, context propagation, customized logging, and utilizing search attributes with the Temporal Go SDK for enhanced Workflow Execution insights.
+  - sdk
+  - go-sdk
+  - developer guide
+description: Monitor your Temporal Application state using Metrics, Tracing, Logging, and Visibility features. Learn about emitting metrics, configuring tracing, context propagation, customized logging, and utilizing search attributes with the Temporal Go SDK for enhanced Workflow Execution insights.
```
Comment (fairlydurable) [inline docs/develop/go/observability.mdx:46]: Not user facing, but Search Attributes is uppercase

File: docs/develop/go/timers.mdx [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ---
 id: timers
-title: Timers - Temporal Go SDK feature guide
-sidebar_label: Durable Timers
+title: Durable Timers - Go SDK
+sidebar_label: Durable timers
```
Comment (fairlydurable) [inline docs/develop/go/timers.mdx:4]:
```suggestion
sidebar_label: Durable Timers
```

File: docs/develop/java/index.mdx [hunk 4]
```diff
@@ -9,127 +9,131 @@ keywords:
...
-The [Workflow messages feature guide](/develop/java/message-passing) shows how to send messages to and read the state of Workflow Executions.
+Send messages to and read the state of Workflow Executions.
 
 - [How to develop with Signals](/develop/java/message-passing#signals)
 - [How to develop with Queries](/develop/java/message-passing#queries)
 - [What is a Dynamic Handler?](/develop/java/message-passing#dynamic-handler)
 - [How to develop with Updates](/develop/java/message-passing#updates)
 
-## Asynchronous Activity Completion
+## [Asynchronous Activity Completion](/develop/java/asynchronous-activity-completion)
```
Comment (fairlydurable) [inline docs/develop/java/index.mdx:64]:
```suggestion
## [Asynchronous Activity completion](/develop/java/asynchronous-activity-completion)
```

File: docs/develop/python/index.mdx [hunk 5]
```diff
@@ -13,122 +13,119 @@ tags:
...
-The [Cancellation feature guide](/develop/python/cancellation) shows how to cancel a Workflow Execution.
+Interrupt a Workflow Execution with a Cancel or Terminate action.
 
-- [Cancel an Activity from a Workflow](/develop/python/cancellation)
+- [Cancel a Workflow](/develop/python/cancellation#cancellation)
+- [Terminate a Workflow](/develop/python/cancellation#termination)
+- [Cancel an Activity from a Workflow](/develop/python/cancellation#cancel-activity)
 
-## Asynchronous Activity Completion
+## [Asynchronous Activity Completion](/develop/python/asynchronous-activity-completion)
```
Comment (fairlydurable) [inline docs/develop/python/index.mdx:70]:
```suggestion
## [Asynchronous Activity completion](/develop/python/asynchronous-activity-completion)
```

File: docs/develop/typescript/entity-pattern.mdx [hunk 6]
```diff
@@ -1,13 +1,16 @@
 ---
 id: entity-pattern
-title: Entity Pattern
+title: Entity Pattern - TypeScript SDK
```
Comment (fairlydurable) [inline docs/develop/typescript/entity-pattern.mdx:3]: Confirm casing. Wasn't sure if "Pattern" or "pattern"

**Review by fairlydurable — CHANGES_REQUESTED** [review]
Pass was not deep: trying to find user-facing errors or pain points in updated material.

Pulled out diffs and ran through spell checker and grammar checker.

Then did visual inspection of files, much of which was rote.

File: docs/develop/python/cancellation.mdx [hunk 7]
```diff
@@ -1,26 +1,60 @@
 ---
 id: cancellation
-title: Cancellation - Python SDK feature guide
-sidebar_label: Cancellation
-description: Learn how to cancel an Activity from a Workflow using the Temporal Python SDK. Ensure proper Heartbeat execution and handling to manage Workflow and Activity cancellations efficiently.
-toc_max_heading_level: 2
+title: Interrupt a Workflow Execution - Python SDK
+sidebar_label: Interrupt a Workflow
+description: interrupt a Workflow Execution using the Temporal Python SDK. Choose between canceling for a graceful stop or terminating for a forceful stop.
```
Comment (napcs) [inline docs/develop/python/cancellation.mdx:5]:
```suggestion
description: Interrupt a Workflow Execution using the Temporal Python SDK. Choose between canceling for a graceful stop or terminating for a forceful stop.
```

File: docs/develop/typescript/entity-pattern.mdx (hunk 6 above)
Comment (napcs) [reply docs/develop/typescript/entity-pattern.mdx:3]:
```suggestion
title: Entity pattern - TypeScript SDK
```

File: docs/develop/go/observability.mdx (hunk 2 above)
Comment (napcs) [reply docs/develop/go/observability.mdx:46]:
```suggestion
description: Monitor your Temporal Application state using Metrics, Tracing, Logging, and Visibility features. Emit metrics, configure tracing, use context propagation, customize logging, and use Search Attributes with the Temporal Go SDK for enhanced Workflow Execution insights.
```

**Review by jsundai — APPROVED** [review]

File: docs/develop/php/index.mdx [hunk 8]
```diff
@@ -9,102 +9,103 @@ keywords:
...
 
 - [Workflow timeouts](/develop/php/failure-detection#workflow-timeouts)
 - [How to set Activity timeouts](/develop/php/failure-detection#activity-timeouts)
 - [How to Heartbeat an Activity](/develop/php/failure-detection#activity-heartbeats)
 
-## Messages
+## [Workflow message passing](/develop/php/message-passing)
 
-The [Workflow messages feature guide](/develop/php/message-passing) shows how to send messages to and read the state of Workflow Executions.
+Send messages to and read the state of Workflow Executions.
```
Comment (jsundai) [inline docs/develop/php/index.mdx:52]:
```suggestion
Send messages to read the state of Workflow Executions.
```
The "to" and "and" is a bit odd there. Ofc that was there before.

**Review by fairlydurable — APPROVED** [review]

**Pull request merged**